rdt/transformers/categorical.py

```diff
@@ -455,13 +457,35 @@ def _fit(self, data):
     def _transform(self, data):
         """Replace each category with its corresponding integer value.
 
+        If a category has not been seen before, a random value is assigned.
+        If no categories have been fitted (and `data` is not empty), raise a `ValueError`.
+
         Args:
             data (pandas.Series):
                 Data to transform.
 
         Returns:
-            numpy.ndarray:
+            pd.Series.
         """
+        # Find the data categories that were not seen during fitting. Convert all nan's
+        # to np.nan to avoid having nan values with different ids.
+        unique_data = {np.nan if pd.isna(x) else x for x in pd.unique(data)}
+        categories = {np.nan if pd.isna(x) else x for x in self.categories_to_values.keys()}
```

Comment:
You can't convert things to `pd.Series`, otherwise it doesn't interpret the the nan's properly:
```python3
set(pd.Series([np.nan])) - set(pd.Series([np.nan]))  # returns {nan}
set([np.nan]) - set([np.nan])  # returns {}
```

Reply:
If we really wanted to vectorize things though, I guess we could do something like this:

```python3
null = pd.isna(data)
unique_data = list(pd.unique(data[~null]))
if null.any():
    unique_data.append(np.nan)
```
Then do the same for `categories` and subtract them.

Reply:
If that's faster by a significant amount I'd suggest going with that